### DISCUSSION TITLE
Changes to IIP52: Add support for Multiple protocols

### DISCUSSION THREAD

**Discussion by AntonAndell** [icon-project/IIPs#67] [General]
To allow XCall to support multiple protocols and also multi-protocol verification of messages. We would like to propose a new IIP to extend IIP52. 
We imagine to achieve this in the simplest way possible by:
Adding some way to configure multiple message centers, implementing the BMC interface, to be added and indexed by a protocol string.
Extend the BTP address format to allow for addresses on the format:
<Protocol1>|<Protocol2>|…|ProtocolN://<ChainId>/<Address>
Then use this new format as 'to' in CSMessageRequest or a new field specifying protocols.

When sending messages, we will send them to all protocols specified in the 'to' address. The same approach will be used for rollbacks. When receiving a message, we will check the protocols specified in the request and only make available for execution  when the message has been received from all specified protocols.

This should introduce no logical difference in usage compared to how XCall is currently used with BTP only. 

@sink772 Do you see any direct red flags on this approach?

Any input and discussion here would be great before drafting a IIP PR.

**Comment by mksong76** [comment]
IIP52 is for defining the XCall service in BTP protocol. But your idea is not in the scope of BTP.

1. New address scheme for the service is to be defined ( this is out of the scope of BTP )
2. New behavior of the service is required

It would be better for you to propose different proposals to extend it.

**Reply by AntonAndell** [reply]
Yes, i agree a new IIP might be the better way to go.
Edited original discussion.
